client/sync-state-rpc/src/lib.rs

```diff
@@ -197,6 +197,9 @@ where
 		let val = serde_json::to_value(&current_sync_state)?;
 		*extension = Some(val);
 
-		chain_spec.as_json(raw).map_err(|e| Error::<Block>::JsonRpc(e).into())
+		chain_spec
+			.as_json(raw)
+			.map(Into::into)
```

Comment:
This seem to work:
```suggestion
			.and_then(|s| serde_json::from_str(&s).map_err(|e| e.to_string()))
```

(thnx Pierre)